snuba/migrations/snuba_migrations/metrics/0007_granularity_10s.py

```diff
@@ -0,0 +1,27 @@
+from typing import Sequence
+
+from snuba.migrations import migration, operations
+from snuba.migrations.snuba_migrations.metrics.templates import create_views, drop_views
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    """
+    Create materialized views for metrics with a 10 second granularity
+    in addition to the existing 60 seconds views.
+
+    The backward migration does *not* delete any data from the destination tables.
+    """
+
+    blocking = False
+
+    def forwards_local(self) -> Sequence[operations.SqlOperation]:
+        return create_views(granularity=10)
+
+    def backwards_local(self) -> Sequence[operations.SqlOperation]:
+        return drop_views(granularity=10)
```

Comment:
I don't think adding all 10s granularities is a good idea as it creates a three views in one migraiton.
The issue is that the atomic unit, when running the migration framework is the individual migration.
The framework does not expose individual statements in a migration. Multiple statements should be in the same migration only if they are supposed to be atomic. Either they all succeed or they are all reversed. 

Three views are largely independent. If one view creation fails the framework won't tell you which one did. It will just tell you this entire migration failed. 

Please split it into three, one per table, each creates a materialized view. Yes it is more verbose but that is  to allow the framework to get into a consistent state after each migration. 

This way will also make the templates simpler to understand.

Also, we are likely one day to split tables into separate clusters. So keeping the migrations separated by use cases (as they previous migrations are) will make things easier there.